File: Problemset/0917-Reverse Only Letters/Stack.py
class Solution:
    def reverseOnlyLetters(self, S: str) -> str:
        stack = []
        ret = ""

        for ch in S:
            if ch.isalpha():
                stack.append(ch)

        for ch in S:
            if ch.isalpha():
                ret += stack.pop()
            else:
                ret += ch

        return ret

File: Problemset/1475-Final Prices With a Special Discount in a Shop/Stack.rs
impl Solution {
    pub fn final_prices(prices: Vec<i32>) -> Vec<i32> {
        let mut stack = vec![];
        let mut ret = vec![0; prices.len()];

        for i in (0..prices.len()).rev() {
            while *stack.last().unwrap_or(&0) > prices[i] {
                stack.pop();
            }
            ret[i] = prices[i] - *stack.last().unwrap_or(&0);
            stack.push(prices[i]);
        }

        ret
    }
}
